src/dist/replication/lib/replica_stub.cpp

```diff
@@ -902,6 +902,83 @@ void replica_stub::on_query_replica_info(const query_replica_info_request &req,
     resp.err = ERR_OK;
 }
 
+void replica_stub::on_query_disk_info(const query_disk_info_request &req,
+                                      /*out*/ query_disk_info_response &resp)
+{
+    resp.err = ERR_OK;
+    for (const auto &dir_node : _fs_manager._dir_nodes) {
+        disk_info info;
+        // app_id == 0 means query all app replica_count
+        if (req.app_id == 0) {
+            for (const auto &holding_primary_replicas : dir_node->holding_primary_replicas) {
+                info.holding_primary_replica_counts[holding_primary_replicas.first] =
+                    static_cast<int>(holding_primary_replicas.second.size());
+            }
+
+            for (const auto &holding_secondary_replicas : dir_node->holding_secondary_replicas) {
+                info.holding_secondary_replica_counts[holding_secondary_replicas.first] =
+                    static_cast<int>(holding_secondary_replicas.second.size());
+            }
+        } else {
+            {
+                zauto_read_lock l(_replicas_lock);
+                bool app_existed = false;
+                for (const auto &replica : _replicas) {
+                    const app_info &info = *(replica.second)->get_app_info();
+                    if (info.app_id == req.app_id) {
+                        app_existed = true;
+                        break;
+                    }
+                }
+                if (!app_existed) {
+                    for (const auto &closing_replica : _closing_replicas) {
+                        const app_info &info = std::get<2>(closing_replica.second);
+                        if (info.app_id == req.app_id) {
```

Comment:
Too many if-statements, could you do some refactor?

Reply:
Yeah, I also think so. but I have it but not good result. I will re-consider it.